lib/index.js

```diff
@@ -267,36 +267,58 @@ King = new Class({
     /**
      * Generates a valid url for a kong request, cycling through one of the hosts for each call
      * @method module:kingkong#request
-     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer ) 
+     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer )
      * @param {String} [id] the id or name of a specific resource
-     * @return {Promise} A promise that will resolve to the result of the request 
+     * @return {Promise} A promise that will resolve to the result of the request
      **/
-    ,request:function _request( action, resource, id, data ){
-        let result = resource_exp.exec( resource )
-          , method = action.toUpperCase()
-          , body   = clone( data )
-          , error
-          , uri
-          ;
+    ,request:function _request( action, resource, id, data, optionalAcceptableStatusCodes ){
+        let requestOptions;
 
-        if(!result){
-            error = new Error('invalid resource type ' + resource);
-            error.name = 'InvalidResourceError';
-            throw error;
+        //  Sometimes we send entire request options object as first parameter in which case
+        //  we don't need to generate the request options.
+        if (typeof action === 'object') {
```

Comment:
Yes, I set `sync` to `false` in my client code. But if the library doesn't recognize HTTP errors returned by Kong (as it doesn't without this PR) and thus doesn't throw exceptions, there is no way to signal to client code that something went wrong. It just seems to work but it doesn't.